libia2/include/ia2.h

```diff
@@ -260,3 +219,10 @@ asm(".macro mov_mixed_pkru_eax pkey0, pkey1\n"
       ia2_stackptrs[i] = stack_start + STACK_SIZE - 8;                         \
     }                                                                          \
   }
+
+
+#ifdef IA2_INIT_COMPARTMENT
+#ifndef LIBIA2_INSECURE
+#include "../ia2.c"
```

Comment:
Still haven't gone through the PR in detail, but the point of moving stuff from ia2.h into a separate library was to avoid including everything it defined multiple times. This may be less of an issue in this case because of the `#ifdef IA2_INIT_COMPARTMENT` but I think we should try to avoid this if possible.

Reply:
I think we need to so we're not taking in a pointer that control the pkey or address. If you have another way to avoid that, I'm definitely open to it.